core/pom.xml

```diff
@@ -357,6 +357,34 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-crypto</artifactId>
     </dependency>
+
+    <!--
+     Testing Hive reflection needs hive on the test classpath only.
```

Comment:
Also, to clarify, this is not adding a dependency to the Spark hive module.  It's adding a dependency to Apache Hive.